File: integrations/claude-agent-sdk/typescript/examples/agentic_chat.ts
/**
 * Agentic chat example - basic configuration.
 *
 * This example shows how to create a basic agentic chat adapter
 * using the Claude Agent SDK integration.
 */

import { ClaudeAgentAdapter } from "@ag-ui/claude-agent-sdk";

/**
 * Create adapter for agentic chat.
 *
 * The adapter configuration supports all ClaudeAgentOptions from the Claude Agent SDK.
 * See: https://platform.claude.com/docs/en/agent-sdk/typescript
 */
export function createAgenticChatAdapter(): ClaudeAgentAdapter {
  return new ClaudeAgentAdapter({
    model: "claude-haiku-4-5",
    systemPrompt: "You are a helpful assistant with access to tools.",
    includePartialMessages: true,
  });
}

File: integrations/claude-agent-sdk/python/examples/agents/agentic_chat.py
"""
Agentic chat agent configuration.

This module provides a factory function for creating an agentic chat adapter.
The adapter supports all ClaudeAgentOptions from the Claude Agent SDK.
"""

from ag_ui_claude_sdk import ClaudeAgentAdapter
from .constants import DEFAULT_DISALLOWED_TOOLS


def create_agentic_chat_adapter(cwd: str) -> ClaudeAgentAdapter:
    """
    Create adapter for agentic chat.
    
    Args:
        cwd: Working directory for conversation state (per-thread).
        
    Returns:
        Configured ClaudeAgentAdapter for general purpose agentic chat.
    """
    return ClaudeAgentAdapter(
        name="agentic_chat",
        description="General purpose agentic chat assistant",
        options={
            "model": "claude-haiku-4-5",
            "system_prompt": "You are a helpful assistant with access to tools.",
            "disallowed_tools": list(DEFAULT_DISALLOWED_TOOLS),
        }
    )
